hooks/post-command

```diff
@@ -1,41 +1,46 @@
-#!/bin/bash
+#!/usr/bin/env bash
 set -euo pipefail
 
-get_os() {
-  local OS=`uname -s`
+error() {
+  >&2 echo "$*"
+  exit 1
+}
 
-  if [ "${OS}" = "Linux" ] ; then
-    if [ -f /etc/alpine-release ] ; then
+print_os() {
+  local OS
+  OS=$(uname -s)
+
+  if [[ "${OS}" = "Linux" ]] ; then
+    if [[ -f /etc/alpine-release ]] ; then
       echo "alpine"
       return
     fi
     echo "linux"
     return
 
-  elif [ "${OS}" == "Darwin" ]; then
+  elif [[ "${OS}" == "Darwin" ]]; then
     echo "macos"
     return
   fi
 
-  echo $(uname -a)
+  echo "${OS}"
 }
 
-OS=$(get_os)
+OS=$(print_os)
 CODECOV_VERSION="${BUILDKITE_PLUGIN_CODECOV_UPLOADER_VERSION:-latest}"
 TMP_DIR="${BUILDKITE_PLUGIN_CODECOV_TMP_DIR:-/tmp}/codecov-buildkite-plugin/${OS}/${CODECOV_VERSION}"
 
 # Reads a list from plugin config into a global result array
 # Returns success if values were read
 plugin_read_list_into_result() {
-  result=()
+  local result=()
```

Comment:
@joscha This is a breaking change. Just see the comment above this function:
`global result array`.
This has already broken the tests and will also break the plugin.
Could fix this and also make sure that the tests pass by running `docker-compose run --rm tests`?
Please ping me again when this is done.

Reply:
Ah, missed that, apologies. I wasn't able to get my docker setup running for some reason. I can try again but something is dodgy. I can revert this and we can try again?